lib/nerve/service_watcher.rb

```diff
@@ -131,17 +145,34 @@ def check_and_report
       is_up = check?
       log.debug "nerve: current service status for #{@name} is #{is_up.inspect}"
 
+      report_succeeded = true
```

Comment:
this initialization seems superfluous

Reply:
this is added here to ensure that check_and_report does not return nil when "is_up == @was_up" and reporter.ping? is successful.

Reply:
so it returns true in this case meaning that zk check is successful and there is no status change

Reply:
to make the intent clear, i would do something like:

```ruby
return true if is_up == @was_up

# if we got here, the state has changed
rest_of_the_code
```

but NBD